nunaliit2-js/src/main/js/nunaliit2/n2.modelUtils.js

```diff
@@ -497,7 +497,579 @@ var ModelIntersect = $n2.Class({
+		this.copyToSchemaDocsByDocId = {};
+		this.copyFromSchemaDocsByDocId = {};
+
+		// Register to events
+		if (this.dispatchService) {
+			f = function(m, addr, dispatcher) {
+				_this._handle(m, addr, dispatcher);
+			};
+
+			this.dispatchService.register(DH, 'modelGetInfo', f);
+			this.dispatchService.register(DH, 'modelGetState', f);
+			this.dispatchService.register(DH, 'modelStateUpdated', f);
+
+			// Initialize state
+			m = {
+				type: 'modelGetState',
+				modelId: this.sourceModelId
+			};
+
+			this.dispatchService.synchronousCall(DH, m);
+			if (m.state) {
+				this._sourceModelUpdated(m.state);
+			}
+		}
+
+		$n2.log(this._classname,this);
+	},
+
+	// Set schemas and join keys used for the transform.
+	_setSchemasToJoin: function(joinInfo) {
+		if (joinInfo) {
+			if (joinInfo.copyToSchema
+				&& typeof joinInfo.copyToSchema === 'string') {
+				this.copyToSchema = joinInfo.copyToSchema;
+			} else {
+				this.copyToSchema = undefined;
+			}
+
+			if (joinInfo.copyFromSchema
+				&& typeof joinInfo.copyFromSchema === 'string') {
+				this.copyFromSchema = joinInfo.copyFromSchema;
+			} else {
+				this.copyFromSchema = undefined;
+			}
+
+			if (joinInfo.copyToJoinKey
+				&& typeof joinInfo.copyToJoinKey === 'string') {
+				this.copyToJoinKey = joinInfo.copyToJoinKey;
+			} else {
+				this.copyToJoinKey = undefined;
+			}
+
+			if (joinInfo.copyFromJoinKey
+				&& typeof joinInfo.copyFromJoinKey === 'string') {
+				this.copyFromJoinKey = joinInfo.copyFromJoinKey;
+			} else {
+				this.copyFromJoinKey = undefined;
+			}
+
+			if (joinInfo.keysToCopy
+				&& $n2.isArray(joinInfo.keysToCopy)
+				&& joinInfo.keysToCopy.length) {
+				this.keysToCopy = joinInfo.keysToCopy;
+
+				// Add doc id key if not specified in keys to copy
+				if (this.keysToCopy.indexOf('doc._id') < 0) {
+					this.keysToCopy.push('doc._id');
+				}
+			} else {
+				this.keysToCopy = undefined;
+			}
+		}
+	},
+
+	_handle: function(m, addr, dispatcher) {
+		var added, docInfo, docInfoKeys, doc;
+		var _this = this;
+		if (m.type === 'modelGetInfo') {
+			if (this.modelId === m.modelId) {
+				m.modelInfo = this._getModelInfo();
+			}
+
+		} else if (m.type === 'modelGetState') {
+			if (this.modelId === m.modelId) {
+				added = [];
+				docInfoKeys = Object.keys(this.docInfosByDocId);
+				docInfoKeys.forEach(function(docId) {
+					docInfo = _this.docInfosByDocId[docId];
+					doc = docInfo.doc;
+					added.push(doc);
+				});
+
+				m.state = {
+					added: added,
+					updated: [],
+					removed: [],
+					loading: this.modelIsLoading
+				};
+			}
+
+		} else if (m.type === 'modelStateUpdated') {
+			// Does it come from our source?
+			if (this.sourceModelId === m.modelId) {
+				this._sourceModelUpdated(m.state);
+			}
+		}
+	},
+
+	_getModelInfo: function() {
+		var modelInfo = {
+			modelId: this.modelId,
+			modelType: this.modelType,
+			parameters: {}
+		};
+		return modelInfo;
+	},
+
+	_cloneDocument: function(doc) {
```

Comment:
Apparently JP thought so as well, because an $n2.deepCopy already exists and does the same thing (learn something new every day). So I've replaced my _cloneDocument() method with the existing deepCopy function.